src/plugins/intel_cpu/src/transformations/snippets/aarch64/pass/snippets_mark_skipped.cpp

```diff
@@ -0,0 +1,311 @@
+        if (auto s = ov::as_type_ptr<ov::op::util::SubGraphOp>(node)) {
+            models.push_back(s->get_function());
+        // Add new multi-body subgraph op here
+        } else if (auto if_op  = ov::as_type_ptr<ov::op::v8::If>(node)) {
+            models.push_back(if_op->get_then_body());
+            models.push_back(if_op->get_else_body());
+        }
+        for (auto& m : models) {
+            for (auto& n : m->get_ops()) {
+                snippets::pass::SetSnippetsNodeType(n, snippets::pass::SnippetsNodeType::SkippedByPlugin);
+                MarkSubgraphOpAsSkipped(n);
+            }
+        }
+    }
+}
+
+auto is_skipped_op(const std::shared_ptr<ov::Node>& op) -> bool {
+    return ov::is_type<ov::op::v0::Constant>(op) ||
+           ov::is_type<ov::op::v0::Parameter>(op) ||
+           ov::is_type<ov::op::v0::Result>(op);
+}
+} // namespace
+
+bool SnippetsMarkSkipped::run_on_model(const std::shared_ptr<ov::Model> &m) {
+    RUN_ON_MODEL_SCOPE(SnippetsMarkSkipped);
+    int channelAxis = DEFAULT_AXIS;
+    for (auto &node : m->get_ordered_ops()) {
+        if (is_skipped_op(node))
+            continue;
+        // We perform this check separately because we mark here only weights path
+        // Matmul itself will be checked further
+        if (isSuitableMatMulWithConstantPath(node)) {
+            auto markup_func = [](Node* node) {
+                SetSnippetsNodeType(node->shared_from_this(), snippets::pass::SnippetsNodeType::SkippedByPlugin);
+            };
+            std::unordered_set<Node*> visited;
+            ov::op::util::visit_constant_path(node->get_input_node_ptr(1), visited, markup_func);
+        }
+        if (isSuitableConvolutionParent(node)) {
+            // Initiate fusing chain
+            SetNodeFusingType(node, NodeFusingType::FusedWithConvolution);
+            channelAxis = DEFAULT_AXIS;
+        } else if (isSuitableBinaryConvolutionParent(node)) {
+            SetNodeFusingType(node, NodeFusingType::FusedWithBinaryConvolution);
+            channelAxis = DEFAULT_AXIS;
+        } else if (isSuitableMiscParent(node)) {
+            channelAxis = DEFAULT_AXIS;
+            SetNodeFusingType(node, NodeFusingType::FusedWithMisc);
+        } else if (isSuitableMatMulParent(node)) {
+            const bool is_fc = isFullyConnected(node);
+            const auto out_rank = node->get_output_partial_shape(0).rank();
+            if (is_fc) {
+                SetNodeFusingType(node, NodeFusingType::FusedWithFC);
+                channelAxis = out_rank.is_static() ? (out_rank.get_length() == 3 ? 2 : 1) : DEFAULT_AXIS;
+            } else {
+                SetNodeFusingType(node, NodeFusingType::FusedWithMatMul);
+                channelAxis = out_rank.is_static() ? out_rank.get_length() - 1 : DEFAULT_AXIS;
+            }
+        } else {
+            for (const auto fusingChainType : getContinuableChains(node)) {
+                if (isSuitableChildForFusingSimple(node, fusingChainType)) {
```

Comment:
Fusings in `ARM` are easier than on `x64`.
They supports `bias` and single activation in advance.

Let's take a look at possible fusings in [GraphOptimizer](https://github.com/xuchen-intel/openvino/blob/ff826fdcda60896f690f2e161ed0497352703990/src/plugins/intel_cpu/src/graph_optimizer.cpp#L58):

1. `FuseConvolutionAndZeroPoints`, `FuseConvMatmulFCDeconvAndDQScales`, - as we know `arm` doesn't support quantization (the whole model (including quantization operations) is executed in floating-point precision).
2. `FuseFCAndWeightsDecompression` is supported only by `x64`. So you can remove the corresponding [checks](https://github.com/xuchen-intel/openvino/blob/8f41607c42792ff3d6cfa410cacb7cb59847b729/src/plugins/intel_cpu/src/transformations/snippets/aarch64/pass/snippets_mark_skipped.cpp#L256) in `SnippetsMarkSkipped`.
3. `FuseConvolutionMatMulDeconvAndBias` - `Conv` nodes, `MatMul` and `Deconv` support `Bias` fusing on `ARM`. However, I don't see these checks in the current impl `SnippetsMarkSkipped` (only [for](https://github.com/xuchen-intel/openvino/blob/8f41607c42792ff3d6cfa410cacb7cb59847b729/src/plugins/intel_cpu/src/transformations/snippets/aarch64/pass/snippets_mark_skipped.cpp#L152) `MatMul`). On `x64` `Bias` are supported [by](https://github.com/xuchen-intel/openvino/blob/dd22a13a29836e5f4d55b49ffb1d9791405594dd/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp#L124) `CanBePerformedAsScaleShift` that is called in `SupportsFusingWithConvolution_Simple`. Thus, at the moment, all biases in models that can be fused into `conv`, `matmul`, `fc`, will be tokenized by Snippets. It will lead to perf degrdadations. I suggest to create new method `isSuitableBias` that can be called for `FusedWithConvolution`, `FusedWithDeconvolution`, `FusedWithMatMul`, `FusedWithFC`. If the Node is possible `bias` (you can take pattern checks from [GraphOptimizer](https://github.com/xuchen-intel/openvino/blob/ff826fdcda60896f690f2e161ed0497352703990/src/plugins/intel_cpu/src/graph_optimizer.cpp#L536)) and if it's bias, propagate the same chain type.
4. `FuseDeconvolutionAndSimpleOperation` - `deconv` on ARM doesn't [support](https://github.com/xuchen-intel/openvino/blob/28d718426780007c8bd5c80b2d6f3506cb33a87d/src/plugins/intel_cpu/src/nodes/deconv.cpp#L331) other fusions (only bias), since `canBePerformedAsScaleShift` returns always `false` for arm.
5. `FuseConvolutionAndSimpleOperationThroughMaxPool` - is supported by current impl.
6. `FuseConvolutionAndSimpleOperation` - `Conv` and `BinaryConv` can fuse only [supported by dnnl](https://github.com/xuchen-intel/openvino/blob/239466ca5d5333af11ee750260215941737383b3/src/plugins/intel_cpu/src/dnnl_extension_utils.cpp#L271) unary ops. At the moment, the current [SupportsFusingWithConvolution_Simple](https://github.com/xuchen-intel/openvino/blob/8f41607c42792ff3d6cfa410cacb7cb59847b729/src/plugins/intel_cpu/src/transformations/snippets/aarch64/pass/snippets_mark_skipped.cpp#L75) contains `Relu`.  Could you please expand the coverage? I understand, that at the moment we support only for `Relu`. So you can add these nodes, comment out them and add the note "remove comment when these nodess will be supported by arm in snippets". Also for `ARM` you check that `defined (OV_CPU_WITH_ACL)` and if it's defined, you need to set new chain type `FusedTerminator` instead of the `FusedWithConvolution` and `FusedWithBinaryConvolution`. Because ACL [supports](https://github.com/xuchen-intel/openvino/blob/ff826fdcda60896f690f2e161ed0497352703990/src/plugins/intel_cpu/src/graph_optimizer.cpp#L1404) only one fusing (excluding bias). `FusedTerminator` is used to mark node that will be fused by plugin and not need to tokenize but fusing chain is interrupted.
7. `FuseFullyConnectedAndSimpleOperation` and `FuseMatMulAndSimpleOperation` - is the same impl as for `Conv` - need to check for [supported by dnnl](https://github.com/xuchen-intel/openvino/blob/239466ca5d5333af11ee750260215941737383b3/src/plugins/intel_cpu/src/dnnl_extension_utils.cpp#L271) unary ops since `MatMul::CanFuse` and `FC::CanFuse` calls `canFuseSimpleOperation` with `isUnarySupportedAsPostOp`. You can just call `SupportsFusingWithConvolution_Simple` here too. However, I don't see any limitations regarding count of possible fused nodes. @alvoron could you please clarify if `MatMul` and `FC` can fuse `Bias` and not-alone Eltwise chains after? Because I see limitation for `Conv` - only one eltwise node after `conv/conv+bias`. We'd like to figure out - has `MatMul` and `FC` the same limitations?
8. `FuseMVNAndSimpleOperation`, `FuseReduceAndSimpleOperation` and `FuseInterpolateAndSimpleOperation` - you removed these checks since these nodes don't support fusings in `arm` - cool!
9. `FuseNormalizeL2AndSimpleOperation` - is already supported as `FusedWithMisc`. Great!


Please, don't hesitate to ask a help with this method. It's very complex method but very needed to avoid many performance regressions!

Reply:
Also I'm not sure about `LSTM` and `AVGPool` in `isSuitableMiscParent`. Could you please verify that these nodes really support fusings on arm?

Reply:
We also have some [tests](https://github.com/xuchen-intel/openvino/blob/c63701060654549cfe5b12c42ae475987d9e467c/src/plugins/intel_cpu/tests/unit/transformations/x64/snipptes_mark_skipped.cpp) for 1SnippetsMarkSkipped` on x64. Could you please add some tests for `arm` as well?

Reply:
Applied. Regarding 6. and 7., I apply FusedTerminator only for FusedWithConvolution, because for now I don't see "!fusedWith.empty()" limitation for Binconv, Matmul and Fullyconnected nodes. @alvoron Please feel free to comment, if it's not true. Thanks Alexandra and Aleksandr!